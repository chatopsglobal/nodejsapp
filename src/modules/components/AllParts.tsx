import * as React from "react";
import * as Queries from "../../services/api/Queries";
import { Query } from "react-apollo";

/**This component is created as an example for
 * Graphql API integration. This will be refined into a common function
 * and will be integrated through redux in subsequent PRs
**/
export class AllParts extends React.Component {

    constructor(props) {
        super(props);
    }
    render() {
        return (<div className="jumbotron">
            <h3>Graphql Sample inetgration results --Listing all parts for the model 9030 </h3>
            <Query query={Queries.partSearchQuery} variables={{ q: "9030" }} >
                {({ loading, error, data }) => {
                    if (loading) return <p>Loading...</p>;
                    if (error) return <p>Error :(</p>;
                    if (!data) return <div>No data!!!</div>;
                    return (
                        <ul className="list-group">
                            Total Count  for parts: {data.totalCount}
                            {data.partSearch.parts.map(parts => <li className="list-group-item" key={parts.id}>
                                part number: {parts.number}
                                <dd> part name: {parts.title}</dd>
                                <dd>price: {parts.price}</dd>
                                <dd>Availability: {parts.availability}</dd>
                                <dd><img src={parts.images} style={{ height: 200, width: 200 }} /> </dd>
                            </li>)}

                        </ul>
                    );
                }}
            </Query>
        </div>);
    }
}
