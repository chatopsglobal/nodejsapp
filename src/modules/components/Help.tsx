import * as React from "react";
import { } from "react-router"

/**This component is created as a sample for
 * routing and page navigation implementation
**/
export class Help extends React.Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <div className="help">
                <div className="">
                    Routed to Help Page
                    <div className="help__caption">
                        Help Page
                    </div>
                </div>

                <br/>

                <div className="help help--modifier">
                    Repair Help
                    <div className="help__caption">
                        Here in the Sears PartsDirect repair help center, you'll find step-by-step instructions and videos for repairing kitchen and laundry appliances, small appliances, and lawn & garden equipment. If you're not sure what the problem is, our descriptions of symptoms can help you pinpoint the solution. If your appliance is flashing an error code, our error code tables explain what the code means and what the problem might be.  And Sears PartsDirect offers millions of replacement parts, making it easy to find the right part for your model.

To get started, scroll through the list below of the repair help categories and select the appliance or tool you need help with.
                    </div>
                </div>
            </div>
        )
    }
}
