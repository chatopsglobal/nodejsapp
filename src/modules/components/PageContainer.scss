.help {
    padding: 20px;
    background: #fff;
    border: 1px solid #eee;
    color: #666;
    malign: 10px;

    // Will be rendered as .help__caption
    &__caption {
        margin-top: 20px;
        padding: 20px;
        background: #002859;
        color: #fff;
    }

    // Will be rendered as .help--modifier
    &--modifier {
        border: 1px solid #eee;
        background: #ebebeb;
    }

    // Will be rendered as .help--modifier help__caption
    &--modifier &__caption {
        color: #666;
        border: 1px solid #666;
        background: #fff;;
    }
}
