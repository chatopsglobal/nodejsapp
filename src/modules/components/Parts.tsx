import * as React from "react";
import { } from "react-router";

/**This component is created as a sample for
 * routing and page navigation implementation
**/

export class Parts extends React.Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <div>
                Routed to Parts Page with param {this.props.match.params.partnum}
            </div>
        )
    }
}
