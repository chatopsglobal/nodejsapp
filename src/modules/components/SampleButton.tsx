import * as React from "react";
import printToConsole from "../../shared/PrintToConsole";

/**This component is created to show a sample
 *  implementation for redux **/
export class SampleButton extends React.Component {

    constructor(props) {
        super(props);
        this.onButtonClick = this.onButtonClick.bind(this);
    }

    onButtonClick() {
        printToConsole("Clicking on Sample Button !");
        this.props.actionButtonClick();
    }

    render() {
        return (<button onClick={this.onButtonClick}>Redux Sample</button>);
    }
}

export default SampleButton;
