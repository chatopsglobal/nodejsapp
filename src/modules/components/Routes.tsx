import * as React from "react";
import {Route, Switch} from "react-router-dom";
import {Help} from "./Help";
import {Parts} from "./Parts";

/**This component contains sample routes that needs
 ** to be updated by the susequent feature related PRs **/

class Routes extends React.Component {

  render() {
    return (
      <div>
        <Switch>
          <Route path="/help" component={Help} />
          <Route path="/parts/:partnum" exact component={Parts} />
        </Switch>
      </div>
    )
  }
}

export default Routes;
