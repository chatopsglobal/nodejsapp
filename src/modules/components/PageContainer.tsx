import * as React from "react";
import { HashRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createStore } from "redux";
import Routes from "./Routes";
import { AllParts } from "./AllParts";
import { ApolloProvider } from "react-apollo";
import { createClient } from "../../services/api/GraphQLUtility"
import ButtonConnector from "../../connectors/SampleButtonConnector";
import { buttonClickReducer } from "../../reducers/SampleReducer";
import { Link } from "react-router-dom";
import './PageContainer.scss';

export let sampleStore = createStore(buttonClickReducer);
const client = createClient();
/**This component is created as a sample page container
 *  which needs to be changed and moved to new folder
 *  by the subsequent feature PRs **/
export class PageContainer extends React.Component {

    // this constructor is necessary to make the props work
    constructor(props) {
        super(props);
    }

    render() {
        return (<React.Fragment>
            <ApolloProvider client={client}>
                <Provider store={sampleStore}>

                    <React.Fragment>
                        <Router>
                            <div>
                                <Link to="/help">Help</Link><br />
                                <Link to="/parts/123">Parts</Link><br />
                                <div className="jumbotron">
                                    <Routes />
                                </div>
                            </div>
                        </Router>
                        <ButtonConnector />
                        <AllParts />
                    </React.Fragment>

                </Provider>
            </ApolloProvider>
        </React.Fragment>);
    }
}
