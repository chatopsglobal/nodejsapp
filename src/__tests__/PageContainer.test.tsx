import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as enzyme from 'enzyme';
import { PageContainer } from "../modules/components/PageContainer";

describe('Page container component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <PageContainer />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
