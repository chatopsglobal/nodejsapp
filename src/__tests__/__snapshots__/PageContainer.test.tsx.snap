// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page container component should render snapshot 1`] = `
Array [
  <div>
    <a
      href="#/help"
      onClick={[Function]}
    >
      Help
    </a>
    <br />
    <a
      href="#/parts/123"
      onClick={[Function]}
    >
      Parts
    </a>
    <br />
    <div
      className="jumbotron"
    >
      <div />
    </div>
  </div>,
  <button
    onClick={[Function]}
  >
    Redux Sample
  </button>,
  <div
    className="jumbotron"
  >
    <h3>
      Graphql Sample inetgration results --Listing all parts for the model 9030 
    </h3>
    <p>
      Loading...
    </p>
  </div>,
]
`;
